hack/go-build.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+set -eu
+
+REPO=github.com/openshift/machine-api-operator
+WHAT=${1:-machine-api-operator}
+GOFLAGS=${GOFLAGS:-}
+GLDFLAGS=${GLDFLAGS:-}
+
+eval $(go env | grep -e "GOHOSTOS" -e "GOHOSTARCH")
+
+GOOS=${GOOS:-${GOHOSTOS}}
+GOARCH=${GOACH:-${GOHOSTARCH}}
```

Comment:
should this be:

```sh
    GOARCH=${GOARCH:-${GOHOSTARCH}}
```

if so, always prefer the form:

```sh
    : ${GOARCH:=${GOHOSTARCH}}
```

which is a) DRY and b) by virtue avoids subsequent typos. The `:` will only apply the value if `GOARCH` is unset.